### PR TITLE
add example of a graphql-layer test

### PR DESCRIPTION
**Pull request by boourns**
@iamkevb

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"os"
+	"testing"
+)
+
+var user *model.User
+
+func TestMain(m *testing.M) {
+	graph.Database = openAndMigrateDatabase("file::memory:?cache=shared")
+
+	user = createUser("bob")
```
Comment (boourns) [inline server/server_test.go:23]: this pattern will eventually bite someone, when you have many test files you'll want all the fixtures created in one spot.  In my bigger app I have a 'test' environment, and when in the 'test' environment an additional migration runs and appends all the fixture data.

Comment (boourns) [reply server/server_test.go:23]: I will add that pattern in here at some point

Comment (iamkevb) [reply server/server_test.go:23]: Would be handy to have database seeds for development as well. Be able to test `user(name: "bob"){id}` w/o running a mutation first.

File: server/server_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	os.Exit(m.Run())
+}
+
+func TestFetchTodos(t *testing.T) {
+	c := client.New(handler.NewDefaultServer(generated.NewExecutableSchema(generated.Config{Resolvers: &graph.Resolver{}})))
+
+	t.Run("Fetch user list", func(t *testing.T) {
+		// response struct must match the shape of the graphql JSON response
+		var resp struct {
+			Users []*model.User `json:"users"`
```
Comment (iamkevb) [inline server/server_test.go:38]: is `json:"users"` needed?

**Review by iamkevb — APPROVED** [review]

File: server/server_test.go (hunk 2 above)
Comment (boourns) [reply server/server_test.go:38]: Absolutely yeah, this shape needs to match the response object for the query and runs through the graphql stack, including a json encode / decode.  It's definitely a bit of a pain to get right when writing new tests.

File: server/server_test.go (hunk 1 above)
Comment (boourns) [reply server/server_test.go:23]: Exactly.  I like chipping away at this, moving patterns from my app into this example, so I'll do the environments pattern I've built sometime soon

**Pull request merged**